app/src/main/java/org/schabi/newpipe/settings/ExoPlayerSettingsFragment.java

```diff
@@ -1,14 +1,45 @@
 package org.schabi.newpipe.settings;
 
+import android.content.SharedPreferences;
 import android.os.Bundle;
 
 import androidx.annotation.Nullable;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceManager;
+import androidx.preference.SwitchPreferenceCompat;
+
+import org.schabi.newpipe.R;
 
 public class ExoPlayerSettingsFragment extends BasePreferenceFragment {
 
     @Override
     public void onCreatePreferences(@Nullable final Bundle savedInstanceState,
                                     @Nullable final String rootKey) {
         addPreferencesFromResourceRegistry();
+
+        final String disableMediaTunnelingAutomaticallyKey =
+                getString(R.string.disabled_media_tunneling_automatically_key);
+        final SwitchPreferenceCompat disableMediaTunnelingPref =
+                (SwitchPreferenceCompat) requirePreference(R.string.disable_media_tunneling_key);
+        final SharedPreferences prefs = PreferenceManager
+                .getDefaultSharedPreferences(requireContext());
+        final boolean mediaTunnelingAutomaticallyEnabled =
+                prefs.getInt(disableMediaTunnelingAutomaticallyKey, -1) == 1;
+        final String summaryText = getString(R.string.disable_media_tunneling_summary);
+        disableMediaTunnelingPref.setSummary(mediaTunnelingAutomaticallyEnabled
+                ? summaryText + getString(R.string.disable_media_tunneling_automatic_info)
+                : summaryText);
+
+        disableMediaTunnelingPref.setOnPreferenceChangeListener((Preference p, Object enabled) -> {
+                    if (((Boolean) enabled).equals(Boolean.FALSE)) {
```

Comment:
I would swap the condition to prevent any `NullPointerException`:

```suggestion
                    if (Boolean.FALSE.equals(((Boolean) enabled))) {
```